agent/plugins/domainjoin/domainjoin_unix_script.go

```diff
@@ -832,20 +831,19 @@ do_domainjoin() {
        # Use username@RemoteTrustedDir (Active Directory Trust) to join
        echo "do_domainjoin(): Found directory in username as username@directory"
     else
-       DOMAIN_USERNAME=${DOMAIN_USERNAME}@${DIRECTORY_NAME}
+       DOMAIN_USERNAME=${DOMAIN_USERNAME}@${REALM}
```

Comment:
if the realm join command needing upper case domain name is specific to RHEL-9, this will need to be targeted only for RHEL-9. We have had realm join with lower case domain name for a while with other Linux distros.

Reply:
I also want to thank you for submitting the PR

Reply:
I have update the PR to restrict the use of REALM in username to only apply to Redhat and Rocky